src/ir/typescript_types.py

```diff
@@ -216,6 +223,84 @@ def get_name(self):
         return 'undefined'
 
 
+class NumberLiteralType(ObjectType):
+    def __init__(self, name, primitive=False):
+        super().__init__(str(name))
+
+    def get_name(self):
+        return self.name;
+
+    def is_string_literal(self):
+        return False
+
+
+class StringLiteralType(ObjectType):
+    def __init__(self, name, primitive=False):
+        super().__init__(name)
+
+    def get_name(self):
+        return '"' + self.name + '"'
+
+    def is_string_literal(self):
```

Comment:
I think this method is not needed (see below about method `get_number_literals`)